frontend/SettingsForm.js

```diff
@@ -0,0 +1,98 @@
+import PropTypes from 'prop-types';
+import React, {Fragment} from 'react';
+import {Field, FieldType, Table} from '@airtable/blocks/models';
+import {
+    Box,
+    Button,
+    FieldPickerSynced,
+    FormField,
+    Heading,
+    SelectButtonsSynced,
+    TablePickerSynced,
+} from '@airtable/blocks/ui';
+
+import {ConfigKeys, IsEnforced} from './settings';
+
+function SettingsForm({setIsSettingsVisible, settings}) {
+    return (
+        <Box
+            flex="1 0 auto"
+            display="flex"
+            flexDirection="column"
+            width="300px"
+            backgroundColor="white"
+            maxHeight="100vh"
+        >
+            <Box
+                flex="auto"
+                display="flex"
+                flexDirection="column"
+                minHeight="0"
+                padding={3}
+                overflowY="auto"
+            >
+                <Heading marginBottom={3}>Settings</Heading>
+                <FormField label="Table">
+                    <TablePickerSynced globalConfigKey={ConfigKeys.TABLE_ID} />
+                </FormField>
+                {settings.table && (
+                    <Fragment>
+                        <FormField
+                            label="URL Field"
+                            description="Must have field type: TEXT or URL"
+                        >
+                            <FieldPickerSynced
+                                table={settings.table}
+                                globalConfigKey={ConfigKeys.URL_FIELD_ID}
+                                allowedTypes={[
+                                    FieldType.SINGLE_LINE_TEXT,
+                                    FieldType.TEXT,
+                                    FieldType.URL,
+                                ]}
+                            />
+                        </FormField>
+                    </Fragment>
+                )}
+                <FormField
+                    label="Enforce Table & URL Field Settings?"
+                    description={`URL Previews will be shown for ${
+                        settings.isEnforced === IsEnforced.YES
+                            ? 'only the specified URL Field in the specified Table'
+                            : 'all valid fields in all available tables'
+                    }.`}
+                >
+                    <SelectButtonsSynced
+                        options={[
+                            {label: 'Yes', value: IsEnforced.YES},
+                            {label: 'No', value: IsEnforced.NO},
+                        ]}
+                        globalConfigKey={ConfigKeys.IS_ENFORCED}
```

Comment:
When I load the Settings page for the first time, neither button for "Enforce Table & URL Field Settings?" is enabled.

Reply:
Seems like a bug in `SelectButtonsSynced`, which is supposed to sync the `value` via the provided `globalConfigKey`